app/components/UI/Swaps/index.js

```diff
@@ -380,6 +387,17 @@ function SwapsAmountView({
     })();
   }, [isTokenInBalances, selectedAddress, sourceToken]);
 
+  useEffect(() => {
+    setAmount('0');
+    setSourceToken(
+      swapsTokens?.find((token) =>
+        toLowerCaseEquals(token.address, initialSource),
+      ),
+    );
+    setDestinationToken(null);
+    setSlippage(AppConstants.SWAPS.DEFAULT_SLIPPAGE);
+  }, [selectedAddress, swapsTokens, initialSource]);
```

Comment:
To ensure that the other dependencies don't unexpectedly reset your states, you may need to track something like `previousSelectedAddress` and do a comparison `selectedAddress !== previousSelectedAddress` before resetting. This guarantees that the only thing that will trigger a reset is a change in account.

Reply:
It seems for doing that I would need to change the `AccountSelector` to receive from `route.params` a function with callback to save the previousSelectedAddress on the `Swaps` component state

I think a better implementation would be on the PreferencesController we have that `previousSelectedAddress` on the state, because this is something that probably we would need across multiple places as we saw on the `TokenRatesController`, what do you think?

Meanwhile on this PR I can update the useEffect for this
```suggestion
// eslint-disable-next-line react-hooks/exhaustive-deps
  }, [selectedAddress);
```